ivy/functional/ivy/experimental/linear_algebra.py

```diff
@@ -1666,3 +1666,61 @@ def dot(
     ivy.array([[-15.28]])
     """
     return current_backend(a, b).dot(a, b, out=out)
+
+
+@handle_nestable
+@to_native_arrays_and_back
+@handle_exceptions
```

Comment:
```suggestion
@handle_nestable
@handle_exceptions
@handle_array_like_without_promotion
@inputs_to_ivy_arrays
@handle_array_function
@handle_device_shifting
```
These are the correct set of decorators that you should be using.

Reply:
@hello-fri-end How do you figure out the right set of decorators for this function?